python-get-all-files-in-directory/create_large_dir.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from pathlib import Path
+
+
+NUM_JUNK_FILES_LARGE = 300
+NUM_JUNK_FILES_MEDIUM = 25
+NUM_JUNK_FILES_SMALL = 5
+
+
+@dataclass
+class Item:
+    name: str
+    children: list[Item] = None
```

Comment:
We can probably use an empty list as a default value here. No need to fight with a `None` value later.

The only complication is that we'll need to import `field` from `dataclass` and set `children: list[Item] = field(default_factory=list)`. But then we can use `if not item.children` in the first `if` test in `create_items()` and skip `if item.children` completely.